dep ensure -add should not depend on Go code

Niondir opened the pull request:
### What does this do / why do we need it?
dep ensure -add should not depend on Go code

### What should your reviewer look out for in this PR?

### Do you need help or clarification on anything?

### Which issue(s) does this PR fix?
fixes #1312

theckman commented:
@Niondir This branch looks to have some outstanding issues that would prevent merging. Do you have the spare cycles to take a look at fixing those things? If you're too busy I _may_ have the spare time to extend the branch (retaining your commits), and open a separate PR.

I wouldn't want to steal your thunder as this seems like a pretty useful change to me. 👍

Niondir commented:
Sure, I will have a look into it.

Niondir commented:
I removed the failing test case, since we added a new one and it seems not to be possible to check for the warning with the current testing framework.

I'm not sure what codeclimate complains about. Some line ending issue with my windows system?

Niondir commented:
Okay almost working, but I have no clue what fails with the AppVeyor test and the `TestSlowVcs` does not like caused by my changes. Any Idea what that could be?

Niondir commented:
What exactly is failing here? Can anyone help?

ibrasho commented:
Seems like a weird network issue. I restarted the AppVeyor job and it seems like it passed now.

ibrasho commented:
@Niondir Could you please resolve the conflicts and rebase on master?

Niondir commented:
I guess no I got it :) Let's hope for CI.

sdboyer commented:
seems like the test failures should be easy to fix with a `go test -update`, it's just failures owing to the introduction of default pruning options.

mvdan commented:
Dep was officially deprecated earlier this year, and the proposal to archive this repository [was accepted](https://github.com/golang/go/issues/38158). As such, I'm closing outstanding issues before archiving the repository. For any further comments, please use the proposal thread on the Go issue tracker. Thanks!
